File: December 2024 Leetcode Solution/December-06.java
class Solution {
    public int maxCount(int[] banned, int n, int maxSum) {
        HashSet<Integer> bannedSet = new HashSet<>();
        for (int b : banned) {
            if (b <= n)
                bannedSet.add(b);
        }

        int ans = 0, sum = 0;

        for (int i = 1; i <= n; i++) {
            if (bannedSet.contains(i))
                continue;

            if (sum + i > maxSum)
                break;

            sum += i;
            ans++;
        }

        return ans;
    }
}

File: December 2024 Leetcode Solution/December-06.rs
use std::collections::HashSet;

impl Solution {
    pub fn max_count(banned: Vec<i32>, n: i32, max_sum: i32) -> i32 {
        let banned_set: HashSet<i32> = banned.into_iter().collect(); 
        let mut ans = 0;
        let mut sum = 0;

        for i in 1..=n {
            if banned_set.contains(&i) {
                continue;
            }

            if sum + i > max_sum {
                break;
            }

            sum += i;
            ans += 1;
        }

        ans
    }
}
